pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -226,7 +227,7 @@ abstract class Analytics {
   Future<void> setTelemetry(bool reportingBool);
 }
 
-class AnalyticsImpl implements Analytics {
+class AnalyticsImplNoEvents implements Analytics {
```

Comment:
I believe that this class needs to be `abstract` because, unless I'm misunderstanding what you're trying to do, it can't implement the full interface from `Analytics`.